packages/temen/src/mapKeys/index.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * "object"와 동일한 값을 가진 객체를 만들고 "iterate"를 통해
+ * "object"의 각 열거형 문자열 키 속성을 이용해 새로운 생성된 키를 생성한합니다.
+ *
+ * 'object'의 키 속성을 'iterate'를 통해 문자열화합니다.
+ * iterate는 세 가지 인수(value, key, object)로 호출됩니다.
+ *
+ * @param {Object} object 반복할 객체
+ * @param {Function} iterate 반복당 호출 함수
+ * @returns {Object} 키 값이 새로 매핑 된 객체
+ *
+ * @example
+ * ```js
+ * mapKeys({ 'a': 1, 'b': 2 }, function(value, key) {
+ *   return key + value;
+ * });
+ * // => { 'a1': 1, 'b2': 2 }
+ * ```
+ */
+
+import getObjectKeys from '../getObjectKeys';
+
+function mapKeys<T>(
+  object: T,
+  iterate: (value: T[keyof T], key: keyof T, object: T) => string | number
+) {
+  return getObjectKeys(object).reduce<Record<string, T[keyof T]>>((acc, key) => {
```

Comment:
```suggestion
  return getObjectKeys(object).reduce<Record<string | number, T[keyof T]>>((acc, key) => {
```
여기는 `iterate` 함수의 반환타입과 맞아야 하지 않을까유? `ReturnType<typeof iterate>` 쓰면 추론될 것 같기도 합니다.

Reply:
https://github.com/Lubycon/lubycon-frontend-libraries/pull/154/commits/7378df1ed6d56a9b177a3f28e82ced01098d5f29 

아직 ReturnType 키워드 익숙하지가 않네유

Reply:
요런 유틸 타입은 생각보다 자주 쓰게되는데도 잘 안 외워지는 것 같아유 ㅋㅋㅋ 저도 맨날 `Extract` 랑 `Exclude` 동작 거꾸로 외우고 있음....